src/ring_theory/polynomial/basic.lean

```diff
@@ -1024,4 +1024,47 @@ begin
   exact ufm_of_gcd_of_wf_dvd_monoid
 end
 
+section
+
+variable [algebra R S]
+
+open algebra finset
+
+lemma ex_mem_adjoin_mul_eq_pow_nat_degree_of_eisenstein_criterion {x : S} {p : R}
```

Comment:
This seems difficult to generalize to an ideal.

Moreover, I don't see a smart way of using the lemma `pow_nat_degree_le_of_aeval_zero_of_eisenstein_criterion_mem_map` from the other file.